### DISCUSSION TITLE
useLazy composable

### DISCUSSION THREAD

**Discussion by farin** [nuxt/nuxt#17418] [Ideas]
I'm not sure how common is this case, but in my project I usually want wait for queries on server side (to get fully rendered page) and not blocking route transition on client side (showing loading placeholders instead)

Nuxt contains lazy option for fetch, but I can't  use it in my case when using GraphQL / urql

Maybe there should be standard composable to wrap any arbitrary action or this topic can be cover by documentation

Now using own composable (in addition accept multuiple ops running them in parallel)
```
export const useLazyQueries = <T extends readonly unknown[] | []>(
  ops: T
): Promise<{ -readonly [P in keyof T]: Awaited<T[P]> }> | T => {
  if (process.server) {
    return Promise.all(ops)
  } else {
    return ops
  }
}
```

**Comment by Flanker72** [comment]
Seems you may use the composable approach suggested in [my article](https://medium.com/@flanker72/nuxt3-complex-solutions-graphql-integration-664e24e48cef) to utilize `useAsyncData` with GraphQL to implement lazy functions. That seems possible with the core Nuxt functionality.
